src/react-native/11-navigation/navigation.md

```diff
@@ -2,23 +2,120 @@
+
+### Why is routing important?
+
+By introducing several screens in our application, it allows us to break down the user interface into smaller, more manageable parts. This makes our application easier to maintain and scale as it grows.
+
+### Deep Linking
+
+Although out of scope for this training, deep linking is a way to navigate to a specific screen or piece of content within your mobile application from an external source. For example, a user might click on a link in an email that opens a specific screen in your app. Deep linking is a powerful tool for user engagement and retention.
+
+The React Navigation library used in this section supports deep linking and has a [guide](https://reactnavigation.org/docs/deep-linking/) on how to set up your application to be notified of incoming links.
+
+### Stacks
+
+In a web application, routing is typically done through anchor tags. Whenever a user clicks on a link, the URL is pushed to the browser history stack. When the user presses the back button, the browser pops the last URL and navigates to the previously visited page.
+
+*Note: A stack is a data structure that follows the Last In, First Out (LIFO) principle.*
+
+React Native does not have a built-in global history stack similar to a web browser. Luckily, there is a library called React Navigation that provides a way to manage navigation in a stack-like manner.
+
+### Navigation Lifecycle
+
+Now that we have an understanding of how navigation works within a stack, let's take a look at the implications of navigation on the component lifecycle.
+
+If we have a stack navigator with two screens, `ScreenA` and `ScreenB`, and navigate to `ScreenA`, then `ScreenA` will mount the component and render. If we then navigate to `ScreenB`, `ScreenA` will be unmounted and `ScreenB` will mount and render. What happens when we navigate back to `ScreenA`? `ScreenB` will unmount the component, but `ScreenA` will *not* remount. Because it stayed in the stack, it remained mounted the entier time.
+
+### Bottom Tab Navigation
+
+Tab navigation pattern is one of the most common navigation patterns used in both Android and iOS applications. It allows users to switch between different screens by tapping on tabs that are typically located at the bottom of the screen.
+
+Often tabs in a mobile application are made up of more than just one screen. Tab navigation is often used in conjunction with stack navigation to allow for more complex navigation patterns. We will cover how to use React Navigation's stack navigation in the next section.
+
+First, let's setup our application to use tab navigation from React Navigation. To do this, we will install the `@react-navigation/bottom-tabs` package:
+
+```bash
+npm install @react-navigation/bottom-tabs
```

Comment:
Same thing here, we can save it for the Setup section.

Reply:
moved to an install step in the setup seciton